### ISSUE TITLE
[Bug]: Concurrency query problem

### ISSUE THREAD

**Issue by zk903956028** [milvus-io/milvus#27383]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.3.0
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When conducting a concurrent test with 8 threads and performing hybrid search with an expression included, there are several instances where the same conditional query fails out of 100 queries.

Please provide the error logs for further assistance.

![企业微信截图_16957294873361](https://github.com/milvus-io/milvus/assets/35655035/dadbcbf9-3faa-4b88-ab9f-3f1a77bc2537)

### Expected Behavior

no thing

### Steps To Reproduce

```markdown
no thing
```

### Milvus Log

no thing

### Anything else?

no thing

**Comment by xiaofan-luan** [comment]
@xiaocai2333 
is this the same issue fixed in 2.3.1?

**Comment by zk903956028** [comment]
> @xiaocai2333 is this the same issue fixed in 2.3.1?

the 2.3.0 i am use!

**Comment by xiaofan-luan** [comment]
confirmed, this should be fixed by 2.3.1. @zk903956028

**Comment by yanliang567** [comment]
@zk903956028 please help to retry on milvus v2.3.1, thx.

/assign @zk903956028 
/unassign
